ravenpy/models/emulators.py

```diff
@@ -641,6 +663,160 @@ def ost2raven(self, ops):
         return self.params(*out)
 
 
+class SACSMA(Raven):
+
+    # Details about SAC-SMA:
+    # https://wiki.ewater.org.au/display/SD41/Sacramento+Model+-+SRG
+
+    identifier = "sacsma"
+    templates = tuple((Path(__file__).parent / "raven-sacsma").glob("*.rv?"))
+
+    params = namedtuple(
+        "SACSMAParams", ", ".join(["par_x{:02}".format(i) for i in range(1, 22)])
+    )
+
+    @dataclass
+    class HRU(HRU):
+        land_use_class: str = "FOREST"
+        veg_class: str = "FOREST"
+        soil_profile: str = "DEFAULT_P"
+        aquifer_profile: str = "[NONE]"
+        terrain_class: str = "[NONE]"
+
+    def __init__(self, *args, **kwds):
+        super().__init__(*args, **kwds)
+        self.rvp = RVP(
+            params=SACSMA.params(*((None,) * len(SACSMA.params._fields))),
+            land_use_classes=(
+                LU("FOREST", impermeable_frac=0.0, forest_coverage=0.02345),
+            ),
+        )
+        self.rvh = RVH(hrus=(SACSMA.HRU(),))
+        self.rvt = RVT(**{k: nc() for k in std_vars})
+        self.rvi = RVI(
+            evaporation="PET_OUDIN", rain_snow_fraction="RAINSNOW_DATA"
+        )  # would like to use RAINSNOW_HBV??
+        self.rvc = RVC(soil0=None, soil2=None, basin_state=BasinIndexCommand())
+        self.rvd = RV(
+            par_soil0_mm=None,
+            par_soil2_mm=None,
+            PAR_BF_LOSS_FRAC=None,
+            POW_X01=None,
+            POW_X02=None,
+            POW_X03=None,
+        )
+
+    def derived_parameters(self):
+        self.rvd["par_soil0_mm"] = self.rvp.params.par_x04 * 1000.0
+        self.rvd["par_soil2_mm"] = self.rvp.params.par_x06 * 1000.0
+        # PAR_BF_LOSS_FRAC = par_x12/(1+par_x12)
+        self.rvd["PAR_BF_LOSS_FRAC"] = self.rvp.params.par_x12
+        # 10.0**self.rvp.params.par_x01  #
+        self.rvd["POW_X01"] = self.rvp.params.par_x01
+        # 10.0**self.rvp.params.par_x02  #
+        self.rvd["POW_X02"] = self.rvp.params.par_x02
+        # 10.0**self.rvp.params.par_x03  #
+        self.rvd["POW_X03"] = self.rvp.params.par_x03
```

Comment:
Why do you need those ?

Reply:
These are derived parameters. They will have a different name in the Ostrich setup and we probably want them to be named the same- no matter if it is in a Raven run or Raven run within the Ostrich calibration setup. Derived parameters exist in ALL other setups too and I tried to follow the logic we are using there too.

Reply:
To be clear, my question is "why do you need copies of parameters with different names?" Could you instead use the original name in the template ? I'm ok with the code as-is if it makes your life easier, it's just a question.

Reply:
In the Ostrich setup they will be named `pow_x01`. So, I like that at least one can see that this the `10**x1` and not actually `x1`. The ranges of parameters we provide is for `x1` while Raven needs `10**x1` as a parameter. I would find it very confusing if it would be called `x1` in the model setup. I am totally fine with how they look like right now. In a very early version of the predecessor of RavenPy we actually passed `10**x1` there and the user had to provide `x1`. At one point we decided that this is very confusing for the users and switched it to the values that are actually used by Raven (i.e. `pow_x01=10**x1`). 

It's indeed very easy for me as it is.

Reply:
But shouldn't it be `self.rvd["POW_X01"] = np.log10(self.rvp.params.par_x01)`

Reply:
No. Because the user provides the value `10**x1` and not `x1`. The problem is that this is a parameter that spans multiple orders of magnitudes (like 0.00001 to 10,000). So, for sampling of this parameter in calibration it is much better to sample in the logarithmic space (like -5 to 4). The first one (like 0.00001 to 10,000) is the one Raven needs (here `pow_x1`) and is hence provided when we run Raven. BUT during calibration and in the Ostrich setup this parameter is the logarithmic one (like -5 to 4) and hence `x1`.

Reply:
Now I get it! Sorry it took so long.

Reply:
No need to be sorry. It is really convoluted because we have Raven itself and then also the Ostrich setup.